Add test for the tumbleweed docker image

Vogtinator opened the pull request:
By setting DOCKER_IMAGE_TEST=1 the test installs the rpm containing the
image and does some basic tests.

Verification run: http://10.160.67.86/tests/25#step/docker_image/20

okurz commented on tests/console/docker_image.pm:
```diff
@@ -0,0 +1,68 @@
+    if (check_var('VERSION', 'Tumbleweed')) {
+        $rpm_name = 'opensuse-image';
+        $image_name = 'opensuse:tumbleweed';
+    }
```
I guess it would be more safe if we fail the test on any other version instead.

okurz commented on tests/console/docker_image.pm:
```diff
@@ -0,0 +1,68 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2018 SUSE LLC
```
Please only the years you worked on this

okurz commented on products/opensuse/main.pm:
```diff
@@ -461,6 +461,11 @@ elsif (get_var('SECURITYTEST')) {
 }
+elsif (get_var('DOCKER_IMAGE_TEST')) {
+    boot_hdd_image;
+    load_docker_tests;
```
Don't we already schedule the docker tests in another case? I don't think we should schedule them multiple times

Vogtinator replied:
This needs to be run in a separate test run to separate image failures from host failures.

The TW docker image will only be released if this particular test passed.

SergioAtSUSE replied:
I agree with @okurz, this will create duplication. It makes sense to have this test separate, but then

1. I will suggest to remove the module from extra tests and rely on this one about docker.
2. Or, in case that the tumbleweed docker image simply needs docker to be installed, then we should split the module **docker** into **install_docker** and **docker**. Then your test module as **tumbleweed_docker_image**.

What do you think?

Vogtinator replied:
I don't agree - IMO we need both tests. `load_docker_tests` is not a prerequisite, but just nice to have as a quick indication whether it's really the image or the host which is broken. If the host is broken, there's no point to test the image either.

Vogtinator commented:
Comments should be addressed now

okurz commented:
Please ensure you have a valid commiter *and* author in all commits.

okurz commented on tests/console/docker_image.pm:
```diff
@@ -0,0 +1,63 @@
+sub run {
+    select_console 'root-console';
+
+    check_var('VERSION', 'Tumbleweed') || die 'Only Tumbleweed uses this test';
```
ok but can we improve the die message to explain the reason? If I would see the message "Only Tumbleweed uses this test" then I would ask "so, what?" :)

Vogtinator replied:
Well, that's literally the only reason. It would work for leap as well, but as the images aren't rolling it doesn't make sense.

The only technical reason is that the image and rpm names would differ and thus the test would fail. That's why the code looked different before, to make that more clear.

SergioAtSUSE replied:
My suggestion:

```perl
die 'At the moment only the tumbleweed docker image is tested';
```

I can imagine reusing this test for Leap 15 and SLE docker images

Vogtinator replied:
That won't happen. Neither Leap nor SLE have docker images as part of snapshots.

okurz commented on tests/console/docker_image.pm:
```diff
@@ -0,0 +1,63 @@
+docker container run --rm -it $image_name /bin/bash; echo DOCKER-\$?- > /dev/$serialdev
+exit 42
+EOF
+
```
If you like instead of using all these file comments you can also specify a fail_message, e.g.

```perl
assert_script_run 'docker load -i /usr/share/suse-docker-images/native/*-image*.tar.xz', fail_message => 'Image could not be loaded';
assert_script_run "docker images $image_name", fail_message => 'Image was not registered';
assert_script_run qq{docker container run --rm $image_name echo "I work" | grep "I work"}, fail_message => 'Could not run executables';
assert_script_run qq{docker container run --rm $image_name grep 'PRETTY_NAME="openSUSE Tumbleweed"' /etc/os-release}, fail_message => 'Could not find openSUSE Tumbleweed';
assert_script_run qq{docker container run --rm $image_name zypper -v ref | grep "All repositories have been refreshed"}, fail_message => 'zypper not working';
```

Vogtinator replied:
That loses the clean indentation though :-/

okurz replied:
you mean the additional blank lines? I consider this a bit too verbose

Vogtinator replied:
No, that each comment/explanation starts at the same line offset. I can remove the blanks.

Vogtinator commented:
> Can you clarify how this PR is related to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/4482/files that is also trying to run the openSUSE Tumbleweed image? I guess the test variable name and documentation can also be a bit clearer on which specific image we are running here, e.g. the image for this snapshot vs. the last published one

This test is supposed to test the image, while the other test tests the host.

SergioAtSUSE commented:
@okurz, I agree, if this PR get merged, I should clean up the docker test module in extra test to not test opensuse:tumbleweed but opensuse:42.3.

okurz commented on tests/console/docker_image.pm:
```diff
@@ -0,0 +1,63 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LINUX GmbH
```
s/SUSE LINUX GmbH/SUSE LLC/

Vogtinator replied:
It's never been SUSE LLC AFAIK, same as in the .spec file headers.
Is there a specific reason it's different here?

okurz replied:
yes, because @coolo said so

Vogtinator replied:
Ok, I'll change it. I'm curious though: Why?

okurz replied:
I guess because "SUSE LLC > SUSE Linux GmbH" and because it's the correct legal entity. Also I think it's "SUSE Linux" and not "SUSE LINUX".

Vogtinator replied:
According to suse.com, only "SUSE LINUX GmbH" and "SUSE PLC" exist, I can't find any LLC anywhere

coolo replied:
They have now an even funnier name :)
"SUSE" and the SUSE logo are trademarks of SUSE IP Development Limited or its subsidiaries or affiliates

Vogtinator replied:
Argh. What's the correct one now?

coolo replied:
use SUSE LLC in this repo - we might mass convert later :)

Vogtinator replied:
Ok, then everything should be ok already.

Vogtinator commented:
Now that a TW snapshot with the awaited changes is there, this is no longer WIP.

I removed most of the blank lines and also changed the message to `die` a bit.

okurz commented:
Can you please update the description and delete the "WIP" reference? Thanks

Otherwise LGTM

Vogtinator commented:
> Can you please update the description and delete the "WIP" reference? Thanks

Done. Isn't the description of the PR normally ignored by the merge?

okurz commented:
hm, somehow I remember wrong then. Well, thanks for questioning my stupid ideas ;)
